src/test/java/io/vertx/junit5/CountingCheckpointTest.java

```diff
@@ -101,4 +103,43 @@ void check_strict_checkpoint() {
       .isInstanceOf(IllegalStateException.class)
       .hasMessage("Strict checkpoint flagged too many times");
   }
+
+  @Test
+  @EnabledIfEnvironmentVariable(named = "SLOW_UNIT_TESTS", matches = ".*")
+  @DisplayName("Check of a lax checkpoint with max value")
+  void check_lax_checkpoint_max_value() {
+    AtomicLong satisfaction = new AtomicLong();
+    CountingCheckpoint checkpoint = CountingCheckpoint.laxCountingCheckpoint(
+        s -> satisfaction.incrementAndGet(), 1);
+
+    assertThat(satisfaction).hasValue(0);
+    checkpoint.flag();
+    assertThat(satisfaction).hasValue(1);
+    for (long i = 0; i < 2 * (long) Integer.MAX_VALUE + 2; i++) {
+      checkpoint.flag();
+    }
+    assertThat(satisfaction).hasValue(1);
+  }
+
+  @Test
+  @EnabledIfEnvironmentVariable(named = "SLOW_UNIT_TESTS", matches = ".*")
+  @DisplayName("Check of a strict checkpoint with max value")
+  void check_strict_checkpoint_max_value() {
```

Comment:
Same here, a better implementation would be to stop counting also for strict checkpoints, and then just thrown an exception whenever `flag()` is being called once the checkpoint has been satisfied.